tests/lib/rules/no-potential-component-option-typo.js

```diff
@@ -0,0 +1,245 @@
+/**
+ * @fileoverview detect if there is a potential typo in your component property
+ * @author IWANABETHATGUY
+ */
+'use strict'
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const rule = require('../../../lib/rules/no-potential-component-option-typo')
+
+const RuleTester = require('eslint').RuleTester
+
+// ------------------------------------------------------------------------------
+// Tests
+// ------------------------------------------------------------------------------
+
+const tester = new RuleTester({
+  parser: require.resolve('vue-eslint-parser'),
+  parserOptions: { ecmaVersion: 2018, sourceType: 'module' }
+})
+tester.run('no-potential-component-option-typo', rule, {
+  valid: [
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+        dat: {},
+        method: {}
+      }
+      </script>
+      `
+    },
+    {
+
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+      }
+      </script>
+      `.trim(),
+      options: [{ presets: ['vue'] }]
+    },
+    // test if give preset and the potentialTypoList length is zero, just for 100% test cover
+    {
+
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+        data() {}
+      }
+      </script>
+      `.trim(),
+      options: [{ presets: ['vue'] }]
```

Comment:
Could you add test cases using a preset other than vue and test cases using multiple presets.